File: services/server/src/main/java/com/yaytsa/server/infrastructure/persistence/entity/UserEntity.java
package com.yaytsa.server.infrastructure.persistence.entity;

import jakarta.persistence.*;
import java.time.OffsetDateTime;
import java.util.UUID;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "users")
public class UserEntity {

  @Id
  @GeneratedValue(strategy = GenerationType.UUID)
  private UUID id;

  @Column(unique = true, nullable = false, length = 50, columnDefinition = "citext")
  private String username;

  @Column(name = "password_hash", nullable = false, length = 255)
  private String passwordHash;

  @Column(name = "display_name", length = 255)
  private String displayName;

  @Column(length = 255, columnDefinition = "citext")
  private String email;

  @Column(name = "is_admin", nullable = false)
  private boolean isAdmin;

  @Column(name = "is_active", nullable = false)
  private boolean isActive = true;

  @CreationTimestamp
  @Column(name = "created_at", nullable = false, updatable = false)
  private OffsetDateTime createdAt;

  @UpdateTimestamp
  @Column(name = "updated_at", nullable = false)
  private OffsetDateTime updatedAt;

  @Column(name = "last_login_at")
  private OffsetDateTime lastLoginAt;
}

File: infra-persistence/auth/src/main/kotlin/dev/yaytsa/persistence/auth/entity/UserEntity.kt
package dev.yaytsa.persistence.auth.entity

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.Id
import jakarta.persistence.Table
import java.time.Instant
import java.util.UUID

@Entity
@Table(name = "users", schema = "core_v2_auth")
class UserEntity(
    @Id
    @Column(name = "id", nullable = false)
    var id: UUID = UUID.randomUUID(),
    @Column(name = "username", nullable = false, columnDefinition = "citext")
    var username: String = "",
    @Column(name = "password_hash", nullable = false)
    var passwordHash: String = "",
    @Column(name = "display_name")
    var displayName: String? = null,
    @Column(name = "email", columnDefinition = "citext")
    var email: String? = null,
    @Column(name = "is_admin", nullable = false)
    var isAdmin: Boolean = false,
    @Column(name = "is_active", nullable = false)
    var isActive: Boolean = true,
    @Column(name = "created_at", nullable = false)
    var createdAt: Instant = Instant.now(),
    @Column(name = "updated_at", nullable = false)
    var updatedAt: Instant = Instant.now(),
    @Column(name = "last_login_at")
    var lastLoginAt: Instant? = null,
    @Column(name = "version", nullable = false)
    var version: Long = 0,
)
